launch/launch/actions/execute_local.py

```diff
@@ -98,6 +98,7 @@ def __init__(
         ]] = None,
         respawn: Union[bool, SomeSubstitutionsType] = False,
         respawn_delay: Optional[float] = None,
+        respawn_max_retries: Optional[int] = None,
```

Comment:
Since we are using a sentinel value for max_retries, this doesn't have to be optional and can always be an int.  This will also simplify some of the code below:

```suggestion
        respawn_max_retries: int = -1,
```